### PR TITLE
ISPN-14003 Document differences between the purge and reindex

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-14003

### CODE REVIEW

**Comment by fax4ever** [comment]
Thank you @wburns, I applied the change.

File: query/src/main/java/org/infinispan/query/impl/IndexStartupRunner.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public static void run(SearchMapping mapping, Indexer indexer, Configuration con
       if (IndexStartupMode.PURGE.equals(startupMode)) {
          mapping.scopeAll().workspace().purge();
       } else if (IndexStartupMode.REINDEX.equals(startupMode)) {
-         indexer.run();
+         indexer.runLocal();
```
[the remark this answers is not in the document]
Comment (wburns) [reply query/src/main/java/org/infinispan/query/impl/IndexStartupRunner.java:18]: > At this point maybe we should make sure that the preload is over by the time we run the mass indexer. 

I wonder if this really matters as we will need to index the store anyways right?

> I don't think that for the use case waiting for the cluster is properly formed makes sense.

I was only thinking regarding this to reduce reindexing the same data multiple times.

For example the first node to be created will index all of the data, Then the second nodes (owners = 2) will index all of the data. Then the third node will index two thirds of the data, the fourth half of it. Then the fifth will index 40% of it. I was just thinking if we optimized for cluster formed then if you had 5 nodes then each would only index 40% of it. But like I said this is a future improvement if possible.

**Comment by wburns** [comment]
Btw looks like another test is affected https://ci.infinispan.org/job/Infinispan/job/PR-10217/2/testReport/junit/org.infinispan.query.api/ManualIndexingTest/ThreadLeakChecker/ However, I am not sure if the purge is properly waited for like the update would be even with your change.

I am integrating as is though as it will reduce the failures and documents the feature a bit more.

File: query/src/main/java/org/infinispan/query/impl/IndexStartupRunner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/impl/IndexStartupRunner.java:18]: Makes sense. We will talk later.

**Comment by fax4ever** [comment]
> Btw looks like another test is affected https://ci.infinispan.org/job/Infinispan/job/PR-10217/2/testReport/junit/org.infinispan.query.api/ManualIndexingTest/ThreadLeakChecker/ However, I am not sure if the purge is properly waited for like the update would be even with your change.
> 
> I am integrating as is though as it will reduce the failures and documents the feature a bit more.

This is the other case we have where the ThreadLeakChecker detects something, occurring when we have a `SegmentListener#topologyChanged` and the query subsystem takes a little to execute purge. Yes, that's a different case we're going to solve with a different PR.

**Comment by fax4ever** [comment]
Thanks for the merge!

**Pull request merged**